File: UI/MyInterviewPreparation/ClientApp/src/Entities/UserSubject.ts
import { BaseRequest } from "./BaseRequest";
import { SubjectStatus } from "./Enums";

export interface UserSubject extends BaseRequest {
    id: number;
    name: string;
    createdDate: string | null;
    updatedDate: string | null;
    subjectStatus: SubjectStatus;
    userId: number;
    systemSubjectId: number | null;
    domainCount: number | null;
}

File: Common/CommonEntities/QuestionAnswer/UserSubject.cs
using CommonEntities.Base;
using static CommonEntities.Enums;

namespace CommonEntities.QuestionAnswer
{
    public class UserSubject : BaseRequest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public SubjectStatus SubjectStatus { get; set; }
        public int UserId { get; set; }
        public int? SystemSubjectId { get; set; }
        public int? DomainCount { get; set; }
    }
}
